ProcessMaker/processmaker
# Process Templates: Edit Process Template in Modeler

## Issue & Reproduction Steps
Ticket: [FOUR-7380](https://processmaker.atlassian.net/browse/FOUR-7380)

As part of the [Process Templates PRD](https://processmaker.atlassian.net/browse/FOUR-7380), this PR adds a feature to enable master builders to edit their templates in the same way they would edit a process. This enhancement will provide a more seamless and familiar experience for master builders when it comes to maintaining the design of their templates.


## Solution
- Added the ability for master builders to edit their templates in the same way they would a process.


## How to Test
1. Ensure you have the 'Edit Template' permission enabled
2. Ensure you have a Process Templates created
3. Navigate to the Templates Tab in the Process Designer
4. Select the Ellipse menu and select the 'Edit Template' option
5. Add changes to the template and save
6. Create a new process using the template
7. Ensure that the SVG view of the template in the 'Create Process' modal reflects the updated template.
8. Run the process. 

**Expected Behavior:**
User is able to update the process template with no errors and create processes from the updated template.

## Code Review Checklist


[FOUR-7380]: https://processmaker.atlassian.net/browse/FOUR-7380?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
